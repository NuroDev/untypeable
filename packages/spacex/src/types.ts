export * from "./_shared/_shared.types";
export * from "./capsules/capsules.types";
export * from "./company/company.types";
export * from "./cores/cores.types";
export * from "./crew/crew.types";
export * from "./dragons/dragons.types";
export * from "./history/history.types";
export * from "./landpads/landpads.types";
export * from "./launches/launches.types";
export * from "./launchpads/launchpads.types";
export * from "./payloads/payloads.types";
export * from "./roadster/roadster.types";
export * from "./rockets/rockets.types";
export * from "./ships/ships.types";
export * from "./starlink/starlink.types";
