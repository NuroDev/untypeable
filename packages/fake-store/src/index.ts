import { initUntypeable } from "untypeable";

import type {
  Cart,
  CartParams,
  Carts,
  CartsParams,
  CreateCartParams,
  DeleteCartParams,
  UpdateCartParams,
  UserCartsParams,
} from "./cart/cart.types";
import type {
  Product,
  ProductsParams,
  Products,
  CreateProductParams,
  UpdateProductParams,
  ProductParams,
  ProductsCategories,
  ProductsCategoriesParams,
  ProductsCategoryParams,
  ProductsCategory,
  DeleteProductParams,
  UpdateProduct,
  CreateProduct,
} from "./products/products.types";

const u = initUntypeable().args<
  "GET" | "POST" | "PUT" | "PATCH" | "DELETE",
  string
>();

const cartRouter = u.router({
  DELETE: {
    /** Delete a cart */
    "/carts/:id": u.input<DeleteCartParams>().output<Cart>(),
  },
  GET: {
    /** Get all carts */
    "/carts": u.input<CartsParams>().output<Carts>(),
    /** Get a single cart */
    "/carts/:id": u.input<CartParams>().output<Cart>(),
    /** Get user carts */
    "/carts/user/:userId": u.input<UserCartsParams>().output<UserCartsParams>(),
  },
  PATCH: {
    /** Update a product */
    "/carts/:id": u.input<UpdateCartParams>().output<Cart>(),
  },
  POST: {
    /** Add a new product */
    "/carts": u.input<CreateCartParams>().output<Cart>(),
  },
  PUT: {
    /** Update a product */
    "/carts/:id": u.input<UpdateCartParams>().output<Cart>(),
  },
});

const loginRouter = u.router({
  DELETE: {},
  GET: {},
  PATCH: {},
  POST: {},
  PUT: {},
});

const productRouter = u.router({
  DELETE: {
    /** Delete a product */
    "/products/:id": u.input<DeleteProductParams>().output<Product>(),
  },
  GET: {
    /** Get all products */
    "/products": u.input<ProductsParams>().output<Products>(),
    /** Get a single product */
    "/products/:id": u.input<ProductParams>().output<Product>(),
    /** Get all categories */
    "/products/categories": u
      .input<ProductsCategoriesParams>()
      .output<ProductsCategories>(),
    /** Get products in a specific category */
    "/products/category/:category": u
      .input<ProductsCategoryParams>()
      .output<ProductsCategory>(),
  },
  PATCH: {
    /** Update a product */
    "/products/:id": u.input<UpdateProductParams>().output<UpdateProduct>(),
  },
  POST: {
    /** Add a new product */
    "/products": u.input<CreateProductParams>().output<CreateProduct>(),
  },
  PUT: {
    /** Update a product */
    "/products/:id": u.input<UpdateProductParams>().output<UpdateProduct>(),
  },
});

const userRouter = u.router({
  DELETE: {},
  GET: {},
  PATCH: {},
  POST: {},
  PUT: {},
});

const router = u
  .router({})
  .merge(productRouter)
  .merge(cartRouter)
  .merge(userRouter)
  .merge(loginRouter);

export type FakeStoreRouter = typeof router;
