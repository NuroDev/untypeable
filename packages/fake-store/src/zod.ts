export * from "./cart/cart.validators";
export * from "./products/products.validators";
