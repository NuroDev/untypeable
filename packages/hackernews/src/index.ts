import { initUntypeable } from "untypeable";

import type { Item, ItemParams } from "./items/items.types";

const u = initUntypeable();

const router = u.router({
  "/v0/item/:id.json": u.input<ItemParams>().output<Item>(),
});

export type HackerNewsRouter = typeof router;

export * from "./items/items.types";
