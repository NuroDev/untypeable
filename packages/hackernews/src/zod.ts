export * from "./items/items.validators";
