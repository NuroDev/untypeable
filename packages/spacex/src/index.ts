import { initUntypeable } from "untypeable";

import type {
  Capsule,
  CapsuleParams,
  Capsules,
} from "./capsules/capsules.types";

const u = initUntypeable();

const router = u.router({
  "/capsules": u.output<Capsules>(),
  "/capsules/:id": u.input<CapsuleParams>().output<Capsule>(),
});

export type SpaceXRouter = typeof router;
