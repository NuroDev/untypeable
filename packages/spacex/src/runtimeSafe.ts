import { initUntypeable } from "untypeable";

import {
  CapsuleSchema,
  CapsuleParamsSchema,
  CapsulesSchema,
} from "./capsules/capsules.validators";
import { CompanySchema } from "./company/company.validators";
import {
  CoreSchema,
  CoreParamsSchema,
  CoresSchema,
} from "./cores/cores.validators";
import {
  CrewSchema,
  CrewMemberSchema,
  CrewMemberParamsSchema,
} from "./crew/crew.validators";
import {
  DragonSchema,
  DragonParamsSchema,
  DragonsSchema,
} from "./dragons/dragons.validators";
import {
  HistoryEventSchema,
  HistoryEventParamsSchema,
  HistoryEventsSchema,
} from "./history/history.validators";
import {
  LandPadSchema,
  LandPadParamsSchema,
  LandPadsSchema,
} from "./landpads/landpads.validators";
import {
  LaunchesV4Schema,
  LaunchesV5Schema,
  LaunchParamsSchema,
  LaunchV4Schema,
  LaunchV5Schema,
} from "./launches/launches.validators";
import {
  LaunchPadSchema,
  LaunchPadParamsSchema,
  LaunchPadsSchema,
} from "./launchpads/launchpads.validators";
import {
  PayloadSchema,
  PayloadParamsSchema,
  PayloadsSchema,
} from "./payloads/payloads.validators";
import { RoadsterSchema } from "./roadster/roadster.validators";
import {
  RocketSchema,
  RocketParamsSchema,
  RocketsSchema,
} from "./rockets/rockets.validators";
import {
  ShipSchema,
  ShipParamsSchema,
  ShipsSchema,
} from "./ships/ships.validators";
import {
  StarlinkSatelliteSchema,
  StarlinkSatelliteParamsSchema,
  StarlinkSatellitesSchema,
} from "./starlink/starlink.validators";

const u = initUntypeable();

export const spaceXSafeRouter = u.router({
  "/v4/capsules": u.output(CapsulesSchema),
  "/v4/capsules/:id": u.input(CapsuleParamsSchema).output(CapsuleSchema),
  "/v4/company": u.output(CompanySchema),
  "/v4/cores": u.output(CoresSchema),
  "/v4/cores/:id": u.input(CoreParamsSchema).output(CoreSchema),
  "/v4/crew": u.output(CrewSchema),
  "/v4/crew/:id": u.input(CrewMemberParamsSchema).output(CrewMemberSchema),
  "/v4/dragons": u.output(DragonsSchema),
  "/v4/dragons/:id": u.input(DragonParamsSchema).output(DragonSchema),
  "/v4/history": u.output(HistoryEventsSchema),
  "/v4/history/:id": u
    .input(HistoryEventParamsSchema)
    .output(HistoryEventSchema),
  "/v4/landpads": u.output(LandPadsSchema),
  "/v4/landpads/:id": u.input(LandPadParamsSchema).output(LandPadSchema),
  "/v4/launches": u.output(LaunchesV4Schema),
  "/v4/launches/latest": u.output(LaunchV4Schema),
  "/v4/launches/next": u.output(LaunchV4Schema),
  "/v4/launches/past": u.output(LaunchesV4Schema),
  "/v4/launches/upcoming": u.output(LaunchesV4Schema),
  "/v4/launches/:id": u.input(LaunchParamsSchema).output(LaunchV4Schema),
  "/v5/launches": u.output(LaunchesV5Schema),
  "/v5/launches/latest": u.output(LaunchV5Schema),
  "/v5/launches/next": u.output(LaunchV5Schema),
  "/v5/launches/past": u.output(LaunchesV5Schema),
  "/v5/launches/upcoming": u.output(LaunchesV5Schema),
  "/v5/launches/:id": u.input(LaunchParamsSchema).output(LaunchV5Schema),
  "/v4/launchpads": u.output(LaunchPadsSchema),
  "/v4/launchpads/:id": u.input(LaunchPadParamsSchema).output(LaunchPadSchema),
  "/v4/payloads": u.output(PayloadsSchema),
  "/v4/payloads/:id": u.input(PayloadParamsSchema).output(PayloadSchema),
  "/v4/roadster": u.output(RoadsterSchema),
  "/v4/rockets": u.output(RocketsSchema),
  "/v4/rockets/:id": u.input(RocketParamsSchema).output(RocketSchema),
  "/v4/ships": u.output(ShipsSchema),
  "/v4/ships/:id": u.input(ShipParamsSchema).output(ShipSchema),
  "/v4/starlink": u.output(StarlinkSatellitesSchema),
  "/v4/starlink/:id": u
    .input(StarlinkSatelliteParamsSchema)
    .output(StarlinkSatelliteSchema),
});
