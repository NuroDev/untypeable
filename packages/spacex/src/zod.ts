export * from "./capsules/capsules.validators";
