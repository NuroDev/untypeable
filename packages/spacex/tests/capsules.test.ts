import { describe, it } from "vitest";

import { CapsuleSchema, CapsulesSchema } from "../src/zod";
import { useTestClient } from "./_client";

describe.concurrent("SpaceX - Capsules", () => {
  const client = useTestClient();

  it("GET - /capsules", async ({ expect }) => {
    const capsules = await client("/capsules");

    expect(capsules).toBeDefined();
    expect(Array.isArray(capsules)).toBe(true);
    expect(capsules.at(0)).toBeDefined();
    expect(capsules.at(-1)).toMatchSnapshot({
      id: "62615d180ec008379be596f1",
      land_landings: 0,
      last_update: null,
      launches: ["6243ade2af52800c6e919255"],
      reuse_count: 0,
      serial: "C212",
      status: "active",
      type: "Dragon 2.0",
      water_landings: 0,
    });

    expect(CapsulesSchema.safeParse(capsules).success).toBe(true);
  });

  it("GET - /capsules/:id", async ({ expect }) => {
    const capsule = await client("/capsules/:id", {
      id: "5e9e2c5bf35918ed873b2664",
    });

    expect(capsule).toBeDefined();
    expect(capsule).toMatchSnapshot({
      id: "5e9e2c5bf35918ed873b2664",
      land_landings: 0,
      last_update: "Hanging in atrium at SpaceX HQ in Hawthorne ",
      launches: ["5eb87cdeffd86e000604b330"],
      reuse_count: 0,
      serial: "C101",
      status: "retired",
      type: "Dragon 1.0",
      water_landings: 1,
    });

    expect(CapsuleSchema.safeParse(capsule).success).toBe(true);
  });
});
