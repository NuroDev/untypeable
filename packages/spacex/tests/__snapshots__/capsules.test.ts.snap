// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SpaceX -  > GET - /capsules/:id 1`] = `
{
  "id": "5e9e2c5bf35918ed873b2664",
  "land_landings": 0,
  "last_update": "Hanging in atrium at SpaceX HQ in Hawthorne ",
  "launches": [
    "5eb87cdeffd86e000604b330",
  ],
  "reuse_count": 0,
  "serial": "C101",
  "status": "retired",
  "type": "Dragon 1.0",
  "water_landings": 1,
}
`;
